### ISSUE TITLE
include full endpoint in traffic status

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1733]
when things are udp, it should include the port

**Comment by cagney** [comment]
For instance:
```
 ipsec trafficstatus
#2: "pool-eastnet-ikev2"[1] 192.1.2.254, type=ESP, add_time=1234567890, inBytes=168, outBytes=168, maxBytes=2^63B, id='C=CA, ST=Ontario, L=Toronto, O=Libreswan, OU=Test Department, CN=north.testing.libreswan.org, E=user-north@testing.libreswan.org', lease=192.0.2.100/32
```
could be changed to something like:
```
#2: "pool-eastnet-ikev2"[1] 192.1.2.254/udp/443, type=ESP, add_time=1234567890, inBytes=168, outBytes=168, maxBytes=2^63B, id='C=CA, ST=Ontario, L=Toronto, O=Libreswan, OU=Test Department, CN=north.testing.libreswan.org, E=user-north@testing.libreswan.org', lease=192.0.2.100/32
```
when `jam_connection_suffix()` sees that the peer is behind a NAT

**Comment by letoams** [comment]
On Tue, 18 Jun 2024, cagney wrote:

> Subject: Re: [libreswan/libreswan] include full endpoint in traffic status (Issue #1733)

>  ipsec trafficstatus
> #2: "pool-eastnet-ikev2"[1] 192.1.2.254, type=ESP, add_time=1234567890, inBytes=168, outBytes=168
> , maxBytes=2^63B, id='C=CA, ST=Ontario, L=Toronto, O=Libreswan, OU=Test Department, CN=north.test
> ing.libreswan.org, ***@***.***', lease=192.0.2.100/32
> 
> could be changed to something like:
> 
> #2: "pool-eastnet-ikev2"[1] 192.1.2.254/udp/443, type=ESP, add_time=1234567890, inBytes=168, outB
> ytes=168, maxBytes=2^63B, id='C=CA, ST=Ontario, L=Toronto, O=Libreswan, OU=Test Department, CN=no
> rth.testing.libreswan.org, ***@***.***', lease=192.0.2.100/32

The idea was to make trafficstatus short and keep it readable and not
put the entire "status of a conn" there. The only reason you see the
IP above is because it is part of the connection name for that instance.

Not because we want to show the traffic selectors in detail.

**Comment by cagney** [comment]
> IP above is because it is part of the connection name for that instance.

Right, `jam_connection_suffix()` is only called for instances.  And I was suggesting it be narrowed further and only add when NAT (or when port changing is allowed such as MOBIKE).   In #1728, for instance, there's nothing in to <<ipsec trafficstatus>> output to alert us or the sysadmin that the peer has being playing with their port.

**Comment by paulwouters** [comment]
I dont think "ipsec trafficstatus" is the place to do this.
I think we agreed to have a json/xml output and people can write their own tools for displaying.

**Comment by cagney** [comment]
> I think we agreed to have a json/xml output and people can write their own tools for displaying.

I don't see how a custom tools that no one uses would not have helped with #1728
Having it display the peer's port on the NAT would.
